packages/client/src/clients/in-app-messages/message-client.ts

```diff
@@ -63,6 +66,9 @@ export class InAppMessagesClient {
       __fetchSource: undefined,
     };
 
+    // Store the last used filters to use in query selector
+    this.fetchOptions = params;
```

Comment:
If custom filters are passed in directly to fetch, we should store that so that we can correctly do the optimistic filtering

Reply:
This doesn't update the default options, it just stores a reference to the last used filters which are used in the query info selector. Alternatively, we could remove `fetchOptions` and rebuild the filter options by parsing the query key.